index/server/main.go

```diff
@@ -172,9 +173,18 @@ func main() {
 		name := c.Param("name")
 		for _, devfileIndex := range index {
 			if devfileIndex.Name == name {
-				bytes, err := pullStackFromRegistry(devfileIndex)
+				var bytes []byte
+				if devfileIndex.Type == indexSchema.StackDevfileType {
+					bytes, err = pullStackFromRegistry(devfileIndex)
+				} else {
+					// Retrieve the sample devfile stored under /registry/samples/<devfile>
+					sampleDevfilePath := path.Join(samplesPath, devfileIndex.Name, devfileName)
+					if _, err = os.Stat(sampleDevfilePath); err == nil {
+						bytes, err = ioutil.ReadFile(sampleDevfilePath)
+					}
+				}
+
 				if err != nil {
-					log.Fatal(err.Error())
```

Comment:
Should we directly remove it or use `log.Print(err.Error())` as errors will be saved in the server log?

Reply:
Yeah, we should still print it

Reply:
Fixed.